pkg/config/autodiscover/autodiscover_targets.go

```diff
@@ -66,6 +70,53 @@ func FindTestTarget(labels []configsections.Label, target *configsections.TestTa
 	}
 
 	target.DeploymentsUnderTest = append(target.DeploymentsUnderTest, FindTestDeployments(labels, target, namespace)...)
+	target.Nodes = GetNodesList()
+}
+
+// GetNodesList Function that return a list of node and what is the type of them.
+func GetNodesList() (nodes map[string]configsections.Node) {
+	nodes = make(map[string]configsections.Node)
+	var nodeNames []string
+	var types []string
+	context := common.GetContext()
+	tester := nodenames.NewNodeNames(common.DefaultTimeout, map[string]*string{"node-role.kubernetes.io/master": nil})
+	test, _ := tnf.NewTest(context.GetExpecter(), tester, []reel.Handler{tester}, context.GetErrorChannel())
+	_, err := test.Run()
+	if err != nil {
+		log.Error("Unable to get node list ", ". Error: ", err)
+		return
+	} else {
+		nodeNames = tester.GetNodeNames()
+		for i := range nodeNames {
+			nodes[nodeNames[i]] = configsections.Node{
+				Name: nodeNames[i],
+				Type: []string{"master"},
+			}
+		}
+	}
+	context = common.GetContext()
+	tester = nodenames.NewNodeNames(common.DefaultTimeout, map[string]*string{"node-role.kubernetes.io/worker": nil})
+	test, _ = tnf.NewTest(context.GetExpecter(), tester, []reel.Handler{tester}, context.GetErrorChannel())
+	_, err = test.Run()
+	if err != nil {
+		log.Error("Unable to get node list ", ". Error: ", err)
+	} else {
+		nodeNames = tester.GetNodeNames()
+		for i := range nodeNames {
+			if _, ok := nodes[nodeNames[i]]; ok {
+				var nod = nodes[nodeNames[i]]
+				nod.Type = append(nod.Type, "worker")
+				nodes[nodeNames[i]] = nod
+
+			} else {
+				nodes[nodeNames[i]] = configsections.Node{
+					Name: nodeNames[i],
+					Type: append(types, "worker"),
```

Comment:
You can also avoid using the "types" variable here. It's always empty since you're never adding anything to it. In fact, you're creating a new Node here with the "worker" type only, so you can just put:
```
Type: []string{"worker"}
```